src/gui/executionlogwidget.cpp

```diff
@@ -32,75 +32,45 @@
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_msgFilterModel(new LogFilterModel(types, this))
 {
     m_ui->setupUi(this);
 
+    LogListView *msgList = new LogListView(this);
+    LogModel *msgModel = new LogModel(this);
+    m_msgFilterModel->setSourceModel(msgModel);
+    msgList->setModel(m_msgFilterModel);
+    connect(msgList, &LogListView::resetModel, [msgModel]() { msgModel->reset(); });
+
+    LogListView *peerList = new LogListView(this);
+    LogPeerModel *peerModel = new LogPeerModel(this);
+    peerList->setModel(peerModel);
+    connect(peerList, &LogListView::resetModel, [peerModel]() { peerModel->reset(); });
+
 #ifndef Q_OS_MACOS
```

Comment:
If possible try moving #ifndef block to the bottom of this function block.

Reply:
done